tensorboard/webapp/settings_link_converter/_redux/settings_link_converter_actions.ts

```diff
@@ -0,0 +1,18 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {createAction, props} from '@ngrx/store';
+import {ShareableSettingState} from './settings_link_converter_types';
+
+export const stateRehydrateFromSharing = createAction(
```

Comment:
I wonder whether even this action and the reducers that handle it might be able to be google-internal. Do you think we can define additional google-internal reducers for Runs and Metrics state? If we don't know the answer then I'd like to try.